isis/src/control/objs/BundleAdjust/BundleAdjust.cpp

```diff
@@ -1896,234 +1880,68 @@ namespace Isis {
-                                                         cameraCoef, &coeffImage(0, index),
-                                                         &coeffImage(1, index));
-          index++;
-        }
-      }
+    if (m_bundleSettings->solveTargetBody()) {
+      observation->computeTargetPartials(coeffTarget, measure, m_bundleSettings, m_bundleTargetBody);
+      observation->computeImagePartials(coeffImage, measure);
     }
 
     // Complete partials calculations for 3D point (latitudinal or rectangular)
-    measureCamera->GroundMap()->GetdXYdPoint(lookBWRTCoord1,
-                                             &coeffPoint3D(0, 0),
-                                             &coeffPoint3D(1, 0));
-    measureCamera->GroundMap()->GetdXYdPoint(lookBWRTCoord2,
-                                             &coeffPoint3D(0, 1),
-                                             &coeffPoint3D(1, 1));
-    measureCamera->GroundMap()->GetdXYdPoint(lookBWRTCoord3,
-                                             &coeffPoint3D(0, 2),
-                                             &coeffPoint3D(1, 2));
-
-    // right-hand side (measured - computed)
-    measuredX = measure.focalPlaneMeasuredX();
-    measuredY = measure.focalPlaneMeasuredY();
 
-    deltaX = measuredX - computedX;
-    deltaY = measuredY - computedY;
+    // Retrieve the coordinate type (latitudinal or rectangular) and compute the partials for
+    // the fixed point with respect to each coordinate in Body-Fixed
+    SurfacePoint::CoordinateType coordType = m_bundleSettings->controlPointCoordTypeBundle();
+    observation->computePoint3DPartials(coeffPoint3D, measure, coordType);
 
-    coeffRHS(0) = deltaX;
-    coeffRHS(1) = deltaY;
+    // right-hand side (measured - computed)
+    observation->computeRHSPartials(coeffRHS, measure);
 
-    // residual prob distribution is calculated even if there is no maximum likelihood estimation
-    double obsValue = deltaX / measureCamera->PixelPitch();
-    m_bundleResults.addResidualsProbabilityDistributionObservation(obsValue);
+    double observationWeight = 1;
+    if (measureCamera->GetCameraType() != 6) {
+      // residual prob distribution is calculated even if there is no maximum likelihood estimation
+      double deltaX = coeffRHS(0);
+      double deltaY = coeffRHS(1);
+      double obsValue = deltaX / measureCamera->PixelPitch();
+      m_bundleResults.addResidualsProbabilityDistributionObservation(obsValue);
 
-    obsValue = deltaY / measureCamera->PixelPitch();
-    m_bundleResults.addResidualsProbabilityDistributionObservation(obsValue);
+      obsValue = deltaY / measureCamera->PixelPitch();
+      m_bundleResults.addResidualsProbabilityDistributionObservation(obsValue);
 
-    observationSigma = 1.4 * measureCamera->PixelPitch();
-    observationWeight = 1.0 / observationSigma;
+      double observationSigma = 1.4 * measureCamera->PixelPitch();
+      observationWeight = 1.0 / observationSigma;
```

Comment:
Observation weighting should be done in the observation. Then, the RHS should be passed to the probability distributions.

Reply:
Weighting has been moved to the observation classes, unless the maximum likelihood option is used in the bundle, in which case some additional weighting is done in `BundleAdjust::computePartials()`